pkg/windows/flannel.go

```diff
@@ -293,15 +293,21 @@ func startFlannel(ctx context.Context, config *FlannelConfig, logPath string) {
 		fmt.Sprintf("--net-config-path=%s", filepath.Join(config.ConfigPath, FlanneldConfigName)),
 	}
 
-	logrus.Infof("Flanneld Envs: %s and args: %v", specificEnvs, args)
-	cmd := exec.CommandContext(ctx, "flanneld.exe", args...)
-	cmd.Env = append(specificEnvs)
-	cmd.Stdout = outputFile
-	cmd.Stderr = outputFile
-	if err := cmd.Run(); err != nil {
-		logrus.Errorf("Flanneld has an error: %v. Check %s for extra information", err, logPath)
+	// We retry running Flanneld 5 times before giving up
+	maxretries := 5
+	for i:=0; i < maxretries; i++ {
```

Comment:
In flannelm we use https://github.com/avast/retry-go/ for retries.
I don't know if it would be worth it her?

Reply:
What specifically are we trying to fix by retrying here? All other prerequisites are polled for readiness before attempting to start the component. Other components that are started in a retry loop don't have a limit on them and are just retried indefinitely while RKE2 is up.

I feel like we should figure out what is causing flannel to fail initially, and poll for that to become come available before starting flannel, rather than artificially limiting the number of restarts.

If we do for some reason need to artificially limit the restarts, shouldn't we `logrus.Fatalf` after the for loop, rather than just leaving rke2 running without any further attempt to start the CNI?

Reply:
The first time flannel runs, it creates the HNS network, which is an overlay flannel network bound with a physical interface. Because of how windows works, that interface then becomes unavailable for some time (according to my measurements from 10s - 25s). If it is closer to the 25s, flanneld dies because it failed to contact kube-api for too long. That's the case that we are trying to fix here

Reply:
One thing we could is: run flannel once and then, if it fails, check for the interface to come back. Once we are sure the interface is back, run flannel again. If it fails a second time, then do the fatal failure. What do you think?

Reply:
Does flannel have an option to do a one-shot startup that will just create the HNS network if necessary, and then exit? We could do that, wait for the network to stabilize, and then start the "permanent" flannel that we expect to not crash?

Reply:
Unfortunately that option does not exist. I have an idea I could try tomorrow in Flannel, hopefully before code freeze